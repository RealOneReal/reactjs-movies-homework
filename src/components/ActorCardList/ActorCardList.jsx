import { Grid } from "@mui/material";
import ActorCard from "../ActorCard";

const ActorCardList = ({ cast }) => {
  return (
    <Grid container spacing={3}>
      {cast.map((card) => (
        <Grid item key={card.id}>
          <ActorCard {...card} />
        </Grid>
      ))}
    </Grid>
  );
};

export default ActorCardList;
