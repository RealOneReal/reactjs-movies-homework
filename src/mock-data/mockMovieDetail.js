const movie = {
  adult: false,
  backdrop_path: "/6VmFqApQRyZZzmiGOQq2C92jyvH.jpg",
  belongs_to_collection: null,
  budget: 200000000,
  genres: [
    {
      id: 18,
      name: "Drama",
    },
    {
      id: 10749,
      name: "Romance",
    },
  ],
  homepage: "",
  id: 597,
  imdb_id: "tt0120338",
  original_language: "en",
  original_title: "Titanic",
  overview:
    "101-year-old Rose DeWitt Bukater tells the story of her life aboard the Titanic, 84 years later. A young Rose boards the ship with her mother and fiancé. Meanwhile, Jack Dawson and Fabrizio De Rossi win third-class tickets aboard the ship. Rose tells the whole story from Titanic's departure through to its death—on its first and last voyage—on April 15, 1912.",
  popularity: 102.676,
  poster_path: "/9xjZS2rlVxm8SFx8kPC3aIGCOYQ.jpg",
  production_companies: [
    {
      id: 4,
      logo_path: "/fycMZt242LVjagMByZOLUGbCvv3.png",
      name: "Paramount",
      origin_country: "US",
    },
    {
      id: 574,
      logo_path: "/iB6GjNVHs5hOqcEYt2rcjBqIjki.png",
      name: "Lightstorm Entertainment",
      origin_country: "US",
    },
    {
      id: 25,
      logo_path: "/qZCc1lty5FzX30aOCVRBLzaVmcp.png",
      name: "20th Century Fox",
      origin_country: "US",
    },
  ],
  production_countries: [
    {
      iso_3166_1: "US",
      name: "United States of America",
    },
  ],
  release_date: "1997-11-18",
  revenue: 2187463944,
  runtime: 194,
  spoken_languages: [
    {
      english_name: "English",
      iso_639_1: "en",
      name: "English",
    },
    {
      english_name: "French",
      iso_639_1: "fr",
      name: "Français",
    },
    {
      english_name: "German",
      iso_639_1: "de",
      name: "Deutsch",
    },
    {
      english_name: "Swedish",
      iso_639_1: "sv",
      name: "svenska",
    },
    {
      english_name: "Italian",
      iso_639_1: "it",
      name: "Italiano",
    },
    {
      english_name: "Russian",
      iso_639_1: "ru",
      name: "Pусский",
    },
  ],
  status: "Released",
  tagline: "Nothing on Earth could come between them.",
  title: "Titanic",
  video: false,
  vote_average: 7.9,
  vote_count: 19931,
};

const credits = {
  id: 597,
  cast: [
    {
      adult: false,
      gender: 2,
      id: 6193,
      known_for_department: "Acting",
      name: "Leonardo DiCaprio",
      original_name: "Leonardo DiCaprio",
      popularity: 23.614,
      profile_path: "/posters/titanic/Leo.jpeg",
      cast_id: 21,
      character: "Jack Dawson",
      credit_id: "52fe425ac3a36847f80179cf",
      order: 0,
    },
    {
      adult: false,
      gender: 1,
      id: 204,
      known_for_department: "Acting",
      name: "Kate Winslet",
      original_name: "Kate Winslet",
      popularity: 14.919,
      profile_path: "/dgIrPQjcdfCnKiGc8ugg5W3GzFX.jpg",
      cast_id: 20,
      character: "Rose DeWitt Bukater",
      credit_id: "52fe425ac3a36847f80179cb",
      order: 1,
    },
    {
      adult: false,
      gender: 2,
      id: 1954,
      known_for_department: "Acting",
      name: "Billy Zane",
      original_name: "Billy Zane",
      popularity: 15.999,
      profile_path: "/9HIubetYWAVLlHNb9aObL0fc0sT.jpg",
      cast_id: 23,
      character: "Cal Hockley",
      credit_id: "52fe425ac3a36847f80179d7",
      order: 2,
    },
    {
      adult: false,
      gender: 1,
      id: 8534,
      known_for_department: "Acting",
      name: "Kathy Bates",
      original_name: "Kathy Bates",
      popularity: 9.516,
      profile_path: "/qZRTzTjV4OC1Ii9a0n8QBS9zMOd.jpg",
      cast_id: 24,
      character: "Molly Brown",
      credit_id: "52fe425ac3a36847f80179db",
      order: 3,
    },
    {
      adult: false,
      gender: 1,
      id: 3713,
      known_for_department: "Acting",
      name: "Frances Fisher",
      original_name: "Frances Fisher",
      popularity: 5.595,
      profile_path: "/tsyGcoXGRNM8KGLpLG3OuxAyxx2.jpg",
      cast_id: 22,
      character: "Ruth Dewitt Bukater",
      credit_id: "52fe425ac3a36847f80179d3",
      order: 4,
    },
  ],
};

export { movie, credits };
