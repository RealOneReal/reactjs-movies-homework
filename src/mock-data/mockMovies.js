const DataFromTMDB = {
  page: 1,
  results: [
    {
      adult: false,
      backdrop_path: "/2va32apQP97gvUxaMnL5wYt4CRB.jpg",
      genre_ids: [14, 28],
      id: 268,
      original_language: "en",
      original_title: "Batman",
      overview:
        'Batman must face his most ruthless nemesis when a deformed madman calling himself "The Joker" seizes control of Gotham\'s criminal underworld.',
      popularity: 61.992,
      poster_path: "/posters/Batman.jpeg",
      release_date: "1989-06-23",
      title: "Batman",
      video: false,
      vote_average: 7.2,
      vote_count: 5807,
    },
    {
      adult: false,
      backdrop_path: "/1BaxMkrN1jUYQcIgRjNOxCr2HNT.jpg",
      genre_ids: [28, 80, 18],
      id: 414906,
      original_language: "en",
      original_title: "The Batman",
      overview:
        "In his second year of fighting crime, Batman uncovers corruption in Gotham City that connects to his own family while facing a serial killer known as the Riddler.",
      popularity: 87.562,
      poster_path: "/posters/Batman.jpeg",
      release_date: "2022-03-03",
      title: "The Batman",
      video: false,
      vote_average: 0,
      vote_count: 0,
    },
    {
      adult: false,
      backdrop_path: "/y9AuabF1SXnn3xZ0tAJLLhv33Uj.jpg",
      genre_ids: [28, 80, 18],
      id: 272,
      original_language: "en",
      original_title: "Batman Begins",
      overview:
        "Driven by tragedy, billionaire Bruce Wayne dedicates his life to uncovering and defeating the corruption that plagues his home, Gotham City.  Unable to work within the system, he instead creates a new identity, a symbol of fear for the criminal underworld - The Batman.",
      popularity: 72.539,
      poster_path: "/posters/BatmanBegins.jpeg",
      release_date: "2005-06-10",
      title: "Batman Begins",
      video: false,
      vote_average: 7.7,
      vote_count: 16728,
    },
    {
      adult: false,
      backdrop_path: "/kU7ZiyeUwcpTkYj3UcxSqGdZmxY.jpg",
      genre_ids: [16, 28],
      id: 618353,
      original_language: "en",
      original_title: "Batman: Death in the Family",
      overview:
        "Tragedy strikes the Batman's life again when Robin Jason Todd tracks down his birth mother only to run afoul of the Joker. An adaptation of the 1988 comic book storyline of the same name.",
      popularity: 166.153,
      poster_path: "/posters/BatmanDeathFamily.jpeg",
      release_date: "2020-10-13",
      title: "Batman: Death in the Family",
      video: false,
      vote_average: 7.3,
      vote_count: 277,
    },
    {
      adult: false,
      backdrop_path: "/eevJuYAitUe6VwFN29aFwzeyeTr.jpg",
      genre_ids: [878, 80, 16, 9648, 28],
      id: 537056,
      original_language: "en",
      original_title: "Batman: Hush",
      overview:
        "A mysterious new villain known only as Hush uses a gallery of villains to destroy Batman's crime-fighting career as well as Bruce Wayne's personal life, which has been further complicated by a  relationship with Selina Kyle/Catwoman.",
      popularity: 41.522,
      poster_path: "/posters/BatmanHush.jpeg",
      release_date: "2019-07-19",
      title: "Batman: Hush",
      video: false,
      vote_average: 7.3,
      vote_count: 561,
    },
    {
      adult: false,
      backdrop_path: "/7WuVUAsWWHt4g11R1HalX0RX2bh.jpg",
      genre_ids: [16, 9648, 28, 80],
      id: 736074,
      original_language: "en",
      original_title: "Batman: The Long Halloween, Part Two",
      overview:
        "As Gotham City's young vigilante, the Batman, struggles to pursue a brutal serial killer, district attorney Harvey Dent gets caught in a feud involving the criminal family of the Falcones.",
      popularity: 184.602,
      poster_path: "/posters/BatmanHush.jpeg",
      release_date: "2021-07-26",
      title: "Batman: The Long Halloween, Part Two",
      video: false,
      vote_average: 7.7,
      vote_count: 200,
    },
    {
      adult: false,
      backdrop_path: "/7x4SNxO5HWArighxxFeet5sn3il.jpg",
      genre_ids: [28, 14],
      id: 364,
      original_language: "en",
      original_title: "Batman Returns",
      overview:
        "While Batman deals with a deformed man calling himself the Penguin, an employee of a corrupt businessman transforms into the Catwoman.",
      popularity: 35.114,
      poster_path: "/posters/Batman.jpeg",
      release_date: "1992-06-19",
      title: "Batman Returns",
      video: false,
      vote_average: 6.8,
      vote_count: 4833,
    },
    {
      adult: false,
      backdrop_path: "/4aJPdkPLnLiiCbfu6WFUQzER3u0.jpg",
      genre_ids: [878, 28, 12, 16],
      id: 321528,
      original_language: "en",
      original_title: "Batman vs. Robin",
      overview:
        'Damian Wayne is having a hard time coping with his father\'s "no killing" rule. Meanwhile, Gotham is going through hell with threats such as the insane Dollmaker, and the secretive Court of Owls.',
      popularity: 34.276,
      poster_path: "/posters/BatmanDeathFamily.jpeg",
      release_date: "2015-04-14",
      title: "Batman vs. Robin",
      video: false,
      vote_average: 7.2,
      vote_count: 655,
    },
    {
      adult: false,
      backdrop_path: "/uS9apevOgs2fuYghf9szOoK1u63.jpg",
      genre_ids: [16, 9648, 28, 80],
      id: 736073,
      original_language: "en",
      original_title: "Batman: The Long Halloween, Part One",
      overview:
        "Following a brutal series of murders taking place on Halloween, Thanksgiving, and Christmas, Gotham City's young vigilante known as the Batman sets out to pursue the mysterious serial killer alongside police officer James Gordon and district attorney Harvey Dent.",
      popularity: 145.804,
      poster_path: "/posters/BatmanHush.jpeg",
      release_date: "2021-06-21",
      title: "Batman: The Long Halloween, Part One",
      video: false,
      vote_average: 7.9,
      vote_count: 247,
    },
    {
      adult: false,
      backdrop_path: "/rpOqHZMNIaI4qP4r7nw1B7oA0mx.jpg",
      genre_ids: [878, 28, 14],
      id: 415,
      original_language: "en",
      original_title: "Batman & Robin",
      overview:
        "Batman and Robin deal with relationship issues while preventing Mr. Freeze and Poison Ivy from attacking Gotham City.",
      popularity: 23.481,
      poster_path: "/posters/Batman.jpeg",
      release_date: "1997-06-20",
      title: "Batman & Robin",
      video: false,
      vote_average: 4.3,
      vote_count: 3868,
    },
    {
      adult: false,
      backdrop_path: "/snlu32RmjldF9b068UURJg8sQtn.jpg",
      genre_ids: [28, 80, 14],
      id: 414,
      original_language: "en",
      original_title: "Batman Forever",
      overview:
        "Batman must battle a disfigured district attorney and a disgruntled former employee with help from an amorous psychologist and a young circus acrobat.",
      popularity: 25.66,
      poster_path: "/mzzNBVwTiiY94xAXDMWJpNPW2US.jpg",
      release_date: "1995-06-16",
      title: "Batman Forever",
      video: false,
      vote_average: 5.4,
      vote_count: 3987,
    },
    {
      adult: false,
      backdrop_path: "/dasEWdFNSobYvY8xWcfz3z02kTu.jpg",
      genre_ids: [28, 12, 14],
      id: 209112,
      original_language: "en",
      original_title: "Batman v Superman: Dawn of Justice",
      overview:
        "Fearing the actions of a god-like Super Hero left unchecked, Gotham City’s own formidable, forceful vigilante takes on Metropolis’s most revered, modern-day savior, while the world wrestles with what sort of hero it really needs. And with Batman and Superman at war with one another, a new threat quickly arises, putting mankind in greater danger than it’s ever known before.",
      popularity: 104.293,
      poster_path: "/posters/BatmanDeathFamily.jpeg",
      release_date: "2016-03-23",
      title: "Batman v Superman: Dawn of Justice",
      video: false,
      vote_average: 5.9,
      vote_count: 15180,
    },
    {
      adult: false,
      backdrop_path: "/jlxDtvmdniVlnGDaDWIrlCiDNpj.jpg",
      genre_ids: [16, 28, 12, 80, 14],
      id: 732450,
      original_language: "en",
      original_title: "Batman: Soul of the Dragon",
      overview:
        "Bruce Wayne faces a deadly menace from his past, with the help of three former classmates: world-renowned martial artists Richard Dragon, Ben Turner and Lady Shiva.",
      popularity: 71.65,
      poster_path: "/uDhnTtSxU5a8DtZdbbin3aZmkmU.jpg",
      release_date: "2021-01-12",
      title: "Batman: Soul of the Dragon",
      video: false,
      vote_average: 7,
      vote_count: 191,
    },
    {
      adult: false,
      backdrop_path: "/djBqGjqojpjnacRsRD2dNnvhQbb.jpg",
      genre_ids: [16, 28, 12],
      id: 251519,
      original_language: "en",
      original_title: "Son of Batman",
      overview:
        "Batman learns he has a violent, unruly pre-teen son, secretly raised by the terrorist group known as The League of Assassins.",
      popularity: 32.007,
      poster_path: "/yHx8OLSKc3VtIBB5lIe90c0fHOX.jpg",
      release_date: "2014-05-13",
      title: "Son of Batman",
      video: false,
      vote_average: 7.1,
      vote_count: 768,
    },
    {
      adult: false,
      backdrop_path: "/nwfpBjtMNFwrGjW2o9HBhMEYCpM.jpg",
      genre_ids: [878, 28, 16],
      id: 366924,
      original_language: "en",
      original_title: "Batman: Bad Blood",
      overview:
        "Bruce Wayne is missing. Alfred covers for him while Nightwing and Robin patrol Gotham City in his stead and a new player, Batwoman, investigates Batman's disappearance.",
      popularity: 34.559,
      poster_path: "/posters/Batman.jpeg",
      release_date: "2016-03-22",
      title: "Batman: Bad Blood",
      video: false,
      vote_average: 7.2,
      vote_count: 577,
    },
    {
      adult: false,
      backdrop_path: "/2JVnUtyUKbQm165cXUF9r5iVPDo.jpg",
      genre_ids: [12, 28, 16, 878],
      id: 408648,
      original_language: "en",
      original_title: "Batman and Harley Quinn",
      overview:
        "Batman and Nightwing are forced to team with the Joker's sometimes-girlfriend Harley Quinn to stop a global threat brought about by Poison Ivy and Jason Woodrue, the Floronic Man.",
      popularity: 29.489,
      poster_path: "/zrqIPpbqCgrs1rS0ltxNicJcuse.jpg",
      release_date: "2017-08-14",
      title: "Batman and Harley Quinn",
      video: false,
      vote_average: 6.2,
      vote_count: 450,
    },
    {
      adult: false,
      backdrop_path: "/jqnbac3oind5VOcAN0mTXyapOGh.jpg",
      genre_ids: [10751, 28, 16, 35, 14],
      id: 324849,
      original_language: "en",
      original_title: "The Lego Batman Movie",
      overview:
        "A cooler-than-ever Bruce Wayne must deal with the usual suspects as they plan to rule Gotham City, while discovering that he has accidentally adopted a teenage orphan who wishes to become his sidekick.",
      popularity: 43.04,
      poster_path: "/posters/Batman.jpeg",
      release_date: "2017-02-08",
      title: "The Lego Batman Movie",
      video: false,
      vote_average: 7.2,
      vote_count: 3865,
    },
    {
      adult: false,
      backdrop_path: "/78WYLdsidjafX5cll2G6fWCR1xB.jpg",
      genre_ids: [10751, 12, 35, 878, 80],
      id: 2661,
      original_language: "en",
      original_title: "Batman",
      overview:
        "The Dynamic Duo faces four super-villains who plan to hold the world for ransom with the help of a secret invention that instantly dehydrates people.",
      popularity: 13.226,
      poster_path: "",
      release_date: "1966-07-30",
      title: "Batman",
      video: false,
      vote_average: 6.3,
      vote_count: 585,
    },
    {
      adult: false,
      backdrop_path: "/wNyhtxSazhYVSG1So8zgGTqVFVo.jpg",
      genre_ids: [28, 16, 80, 18],
      id: 382322,
      original_language: "en",
      original_title: "Batman: The Killing Joke",
      overview:
        "As Batman hunts for the escaped Joker, the Clown Prince of Crime attacks the Gordon family to prove a diabolical point mirroring his own fall into madness.",
      popularity: 35.507,
      poster_path: "/posters/Batman.jpeg",
      release_date: "2016-07-24",
      title: "Batman: The Killing Joke",
      video: false,
      vote_average: 6.6,
      vote_count: 1349,
    },
    {
      adult: false,
      backdrop_path: "/bAM1hVZEQRo1ZujryQ47bOtxEe8.jpg",
      genre_ids: [53, 16, 28, 80],
      id: 242643,
      original_language: "en",
      original_title: "Batman: Assault on Arkham",
      overview:
        "Batman works desperately to find a bomb planted by the Joker while Amanda Waller sends her newly-formed Suicide Squad to break into Arkham Asylum and recover vital information stolen by the Riddler.",
      popularity: 33.738,
      poster_path: "",
      release_date: "2014-08-12",
      title: "Batman: Assault on Arkham",
      video: false,
      vote_average: 7.4,
      vote_count: 824,
    },
  ],
  total_pages: 7,
  total_results: 128,
};

const genres = [
  {
    id: 28,
    name: "Action",
  },
  {
    id: 12,
    name: "Adventure",
  },
  {
    id: 16,
    name: "Animation",
  },
  {
    id: 35,
    name: "Comedy",
  },
  {
    id: 80,
    name: "Crime",
  },
  {
    id: 99,
    name: "Documentary",
  },
  {
    id: 18,
    name: "Drama",
  },
  {
    id: 10751,
    name: "Family",
  },
  {
    id: 14,
    name: "Fantasy",
  },
  {
    id: 36,
    name: "History",
  },
  {
    id: 27,
    name: "Horror",
  },
  {
    id: 10402,
    name: "Music",
  },
  {
    id: 9648,
    name: "Mystery",
  },
  {
    id: 10749,
    name: "Romance",
  },
  {
    id: 878,
    name: "Science Fiction",
  },
  {
    id: 10770,
    name: "TV Movie",
  },
  {
    id: 53,
    name: "Thriller",
  },
  {
    id: 10752,
    name: "War",
  },
  {
    id: 37,
    name: "Western",
  },
];

// todo: should do correct implementation in Redux in future of changing index for string value

DataFromTMDB.results.map((film, indexFilm) => {
  return film.genre_ids.map((categorie, indexCategorie) => {
    const filter = genres.find((genre) => genre.id === categorie);
    DataFromTMDB.results[indexFilm].genre_ids[indexCategorie] = filter.name;
    return filter.name;
  });
});

export default DataFromTMDB;
