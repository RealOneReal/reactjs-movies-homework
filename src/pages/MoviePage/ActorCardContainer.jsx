import { Box } from "@mui/system";
import ActorCardList from "../../components/ActorCardList/ActorCardList";
import { credits } from "../../mock-data/mockMovieDetail";

const ActorCardContainer = () => {
  const cast = credits.cast?.slice(0, 5);
  return (
    <Box sx={{ mt: 2 }}>
      <ActorCardList {...{ cast }} />
    </Box>
  );
};

export default ActorCardContainer;
