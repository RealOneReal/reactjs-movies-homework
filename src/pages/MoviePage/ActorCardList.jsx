import { Grid } from "@mui/material";
import ActorCard from "../../components/ActorCard/ActorCard";
import { credits } from "../../mock-data/mockMovieDetail";

const ActorCardList = () => {
  const cast = credits.cast?.slice(0, 5);

  return (
    <Grid container spacing={3} sx={{ mt: 2 }}>
      {cast &&
        cast.map((card) => (
          <Grid item key={card.id}>
            <ActorCard {...card} />
          </Grid>
        ))}
    </Grid>
  );
};

export default ActorCardList;
