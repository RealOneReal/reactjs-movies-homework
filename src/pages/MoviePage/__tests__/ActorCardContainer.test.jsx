import renderer from "react-test-renderer";
import ActorCardContainer from "../ActorCardContainer";
import ActorCardList from "../../../components/ActorCardList/ActorCardList";
import { credits } from "../../../mock-data/mockMovieDetail";

jest.mock("../../../mock-data/mockMovieDetail", () => {
  const credits = {
    cast: [
      "Titanic",
      "Batman",
      "Brat 1",
      "Brat 2",
      "Brat 3",
      "Brat 4",
      "Brat 5",
    ],
  };
  return { credits };
});
jest.mock("../../../components/ActorCardList/ActorCardList", () => {
  return "ActorCardList";
});

describe("ActorCardContainer", () => {
  it("should be render", () => {
    const tree = renderer.create(<ActorCardContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
