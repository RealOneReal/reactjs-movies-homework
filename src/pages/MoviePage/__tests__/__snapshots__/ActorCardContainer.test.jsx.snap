// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorCardContainer should be render 1`] = `
<div
  className="MuiBox-root css-1yuhvjn"
>
  <ActorCardList
    cast={
      Array [
        "Titanic",
        "Batman",
        "Brat 1",
        "Brat 2",
        "Brat 3",
      ]
    }
  />
</div>
`;
